operations.py

```diff
@@ -38,12 +64,341 @@ def kleene_star(automaton1: Automaton) -> Automaton:
+
+    states = set()
+
+    state = State("empty")
+    for alph in automaton.alphabet:
+        state.add_transition("empty", alph)
+    states.add(state)
+
+    for s in sorted(automaton.states):
+        state = State(s.name)
+        print(f"In state {state}")
+        for alph in sorted(automaton.alphabet):
+            found = False
+            duplicated = False
+            target = "empty"
+            for trans in s.transitions:
+                if trans.symbol == alph:
+                    if not found:
+                        found = True
+                        target = trans.target
+                    else:                                      
+                        duplicated = True
+                        target = clean_target(target + trans.target, automaton.states)
+
+            #TODO: Check for the original states instead of len
+            if duplicated and len(target) > len(alph):
+                print(f"Adding State {target}")
+                states.add(State(target))
+
+            print(f"Adding transition {state.name}:{alph}>{target}")
+
+            state.add_transition(target, alph)
+        
+        states.add(state)
+
+    repeat = True
+    #Check for the new states created
+    while repeat:
+        repeat = False
+#        print(f"Starting repeat loop")
+        for st in sorted(states - automaton.states):
+#            print(f"State {st.name}")
+            if st.name == "empty" or st.worked:
+               continue
+            
+            st.worked = True
+            my_dict = get_new_transitions(automaton, st)
+
+#            print(f"State {st.name} {my_dict}")
+            for alph in automaton.alphabet:
+                #new_state = ''.join(sorted(set(my_dict[alph]))
+                new_state = my_dict.get(alph, None)
+                if new_state != None:
+                    st.add_transition(new_state, alph)
+                    len1 = len(states)
+                    states.add(State(new_state))
+                    len2 = len(states)
+#                print(f"Adding new state: {new_state}, previous len {len1} after adding {len2}")
+                    if len1 < len2:
+                        repeat = True
+                else:
+                    st.add_transition("empty", alph)
+                    print(f"ERROR: dictionary {my_dict} doesn't have values for key {alph} for state {st.name}")       
+
+            
+    accept = set()
+    for a in automaton.accept:
+        for ast in states:
+            if a in ast.name:
+                accept.add(ast.name)
+
+#    print("Original Automaton")
+#    print(automaton.render())
+
+    active_states = remove_unused_states(automaton.initial, states)
+#    active_states = states
+#    print(f"States {states}")
+#    print(f"Active states {active_states}")
+    #Testing:
+    result = Automaton(active_states,automaton.initial,accept, automaton.alphabet)
+
+#    print("Removing duplicated transitions")
+    print(result.render())
```

Comment:
Printing different automaton renders to stdout might be confusing for users, lets add a logging package for debugging purposes and add a `--verbosity` option as an argument